consensus/istanbul/backend/api.go

```diff
@@ -41,19 +41,23 @@ type API struct {
 	istanbul *backend
 }
 
-// GetSnapshot retrieves the state snapshot at a given block.
+// GetSnapshot retrieves the state snapshot at a given block. The namespace is istanbul.
 func (api *API) GetSnapshot(number *rpc.BlockNumber) (*Snapshot, error) {
 	// Retrieve the requested block number (or current if none requested)
 	var header *types.Header
 	if number == nil || *number == rpc.LatestBlockNumber {
 		header = api.chain.CurrentHeader()
+	} else if *number == rpc.PendingBlockNumber {
+		logger.Trace("Cannot get snapshot of the pending block.", "number", number)
+		return nil, errPendingNotAllowed
 	} else {
 		header = api.chain.GetHeaderByNumber(uint64(number.Int64()))
 	}
 	// Ensure we have an actually valid block and return its snapshot
 	if header == nil {
-		return nil, errUnknownBlock
+		return nil, errUnknownBlock // return nil if block is not found.
```

Comment:
What about removing the comment? I think the code itself is clear to understand.

Reply:
Removed the comment.
Thank you!